File: POBJ/Csharp/Ex_4_Reprise_Cpp/Ex_4_Reprise_Cpp/Rectangle.cs
using System;

public class Rectangle : FormesGeometriques
{
    // Constructeur
    public Rectangle()
    {
        Console.WriteLine("Constructeur du Rectangle");
    }

    public override double CalculerSurface(double largeur, double hauteur)
    {
        Surface = largeur * hauteur;
        return Surface;
    }

    public override double CalculerPerimetre(double largeur, double hauteur)
    {
        Perimetre = 2 * (largeur + hauteur);
        return Perimetre;
    }
}

File: POBJ/CPP/Exercices/EX_4/EX_4/Rectangle.cpp
#include "Rectangle.h"
#include <cmath>


// Constructeur
    Rectangle::Rectangle()
    {
    std::cout << "Constructeur du Rectangle" << std::endl;
    }

//Destructeur
    Rectangle::~Rectangle()
{
    std::cout << "Destructeur de Rectangle" << std::endl;
}

double Rectangle::CalculerSurface(double largeur, double hauteur)
{
    Surface = largeur * hauteur;
    return Surface;
}

double Rectangle::CalculerPerimetre(double largeur, double hauteur)
{
    Perimetre = 2 * (largeur + hauteur);
    return Perimetre;
}
